lib/src/platform.dart

```diff
@@ -0,0 +1,129 @@
+import 'pubspec.dart';
+
+abstract class KnownPlatforms {
+  static const String browser = 'browser';
+  static const String console = 'console';
+  static const String flutter = 'flutter';
+  static const String mirrors = 'mirrors';
+
+  static const String angular = 'angular';
+}
+
+class PlatformSummary {
+  final Platform package;
+  final Map<String, Platform> libraries;
+
+  PlatformSummary(this.package, this.libraries);
+
+  bool get hasConflict =>
+      package.hasConflict ||
+      libraries.values.any((p) => p.hasConflict) ||
+      libraries.values.any((p) => p.excludes
+          .any((s) => package.only.contains(s) || package.uses.contains(s)));
+}
+
+class Platform {
+  final List<String> only;
```

Comment:
during the analysis it is a `Set`, but after the results are done, it is more convenient to use `List` for storing in JSON, validating it in a test or displaying it in predictable order.